pkgs/build-support/docker/default.nix

```diff
@@ -681,6 +681,38 @@ rec {
     in
     result;
 
+  # Merge the tarballs of images built with buildImage into a single
+  # tarball that contains all images. Running `docker load` on the resulting
+  # tarball with load the images into the docker daemon.
+  mergeImages = images: runCommand "merge-docker-images"
+    {
+      inherit images;
+      nativeBuildInputs = [ pigz jq ];
+    } ''
+    mkdir image inputs
+    # Extract images
+    repos=()
+    manifests=()
+    for item in $images; do
+      name=$(basename $item)
+      mkdir inputs/$name
+      tar -I pigz -xf $item -C inputs/$name
+      repos+=(inputs/$name/repositories)
+      manifests+=(inputs/$name/manifest.json)
+    done
+    # Copy all layers from input images to output image directory
+    cp -R --no-clobber inputs/*/* image/
+    # Merge repositories objects and manifests
+    jq -s add "''${repos[@]}" > repositories
+    jq -s add "''${manifests[@]}" > manifest.json
+    # Replace output image repositories and manifest with merged versions
+    mv repositories image/repositories
+    mv manifest.json image/manifest.json
+    # Create tarball and gzip
+    tar -C image --hard-dereference --sort=name --mtime="@$SOURCE_DATE_EPOCH" --owner=0 --group=0 --xform s:'^./':: -c . | pigz -nT > $out
```

Comment:
> `--owner=0` etc

This changes the ownership info of the original images. This can be solved by running in fakeroot.
We should probably keep the mtimes from the original images as well.

Reply:
I thought for a minute `fakeroot` was already introduced in `buildLayeredImage`, but https://github.com/NixOS/nixpkgs/pull/116749 isn't merged yet. It may already be useful as an example of using fakeroot though.
For testing you could use `buildImage` with `runAsRoot` for the time being.

Reply:
I don't really understand all the implications of permissions and the `--xform` flag. I simply copied how it was done in the `buildImage` command. I'll try to do this in a `fakeroot` and keep the modification times from the source images.